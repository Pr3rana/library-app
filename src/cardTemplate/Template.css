.bookCard{
    flex: 15%;
    max-width: 20%;
    text-align: center;
    font-family: Georgia, 'Times New Roman', Times, serif;
    font-size: 2rem;
    font-weight: bold;
    background-color: #D0840B;
    color: antiquewhite;
    margin: .5rem;
    padding: 2rem;
    box-shadow: 2px 5px #6a7a8f;
}
#header{
    text-align: center !important;
}
.author{
  font-size: 1rem;
  font-size: 1rem;
  height: 3.5rem;
  border-top: 1px dashed;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}
.summary{
  font-size: .8rem;
  font-weight: lighter;
  padding: 5px 10px 5px 10px;
  height: 7rem;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 8;
  -webkit-box-orient: vertical;
}
.title{
  font-style: oblique;
  font-size: 1rem;
  height: 3.5rem;
  border-bottom: 1px dashed;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
}
#mainWrapper{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin-top: 2rem;
}

@media (max-width: 800px) {
    .bookCard {
      flex: 28%;
      max-width: 36%;
    }
  }
  
  @media (max-width: 600px) {
    .bookCard {
      flex: 100%;
      max-width: 100%;
    }
  }
