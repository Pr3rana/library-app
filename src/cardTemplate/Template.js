import React, { useState } from 'react';
import './Template.css';
import SearchBar from '../searchBar/SearchBar';

/* Template for header */
function Header(){
    return <h3 id="header">Unibuddy Library</h3>
}

/* Template for book card */
export function BookCard({title, summary,author}){
    return(
            <div className="bookCard"> 
                <p className="title">{title}</p>
                <p className="summary">{summary}</p>
                <p className="author">By: {author}</p>
            </div>
    )
}
/* Embedding all components to render the final view */
function Template(){
    const [books, setSelectBooks] = useState([]);
    return (
        <div>
            <Header />
            <SearchBar onSubmit={(book) => {
                /* updating the selected autosuggest result to books */
                setSelectBooks([...books, book])
            }}/>
            <div id="mainWrapper">
                {/* mapping the final result of the search to show the cards  */}
            {books.map((book, index)=>
                <BookCard 
                key={index} 
                title={book.title} 
                summary={book.summary} 
                author={book.author}
            />
            )}
            </div>
        </div>
            
    );
}
export default Template;
