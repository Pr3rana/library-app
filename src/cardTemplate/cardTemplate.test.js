import React from 'react';
import renderer from 'react-test-renderer';
import Template, {BookCard} from './Template'

it('renders correctly', () => {
  const tree = renderer
    .create(<Template />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders correctly', () => {
    const tree = renderer
      .create(<BookCard title="Test" author="Prerna" summary="true"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
