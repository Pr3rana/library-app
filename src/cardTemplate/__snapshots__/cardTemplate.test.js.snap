// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3
    id="header"
  >
    Unibuddy Library
  </h3>
  <div
    id="searchBar"
  >
    <input
      id="myInput"
      name="MyLibraray"
      onChange={[Function]}
      placeholder="Search"
      required={true}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      <i
        className="fa fa-search"
      />
    </button>
  </div>
  <div
    id="mainWrapper"
  />
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="bookCard"
>
  <p
    className="title"
  >
    Test
  </p>
  <p
    className="summary"
  >
    true
  </p>
  <p
    className="author"
  >
    By: 
    Prerna
  </p>
</div>
`;
